R/parse_obs_types.R

```diff
@@ -0,0 +1,25 @@
+#' Parse observation types to simulation code
+#'
+#' @param obs specified observation object, e.g. `list(variable="CONC", scale="1")` or `list(cmt=1, scale="V/1000.0")`.
+parse_obs_types <- function(obs) {
+  if(is.null(obs$scale)) obs$scale <- "1.0"
+  if(length(obs$variable) == 1) {
+    tmp <- paste0("      obs.insert(obs.end(), ", obs$variable, "/(", obs$scale,")); ")
+  } else {
+    if(length(obs$scale) < length(obs$variable)) {
+      obs$scale <- rep(obs$scale[1], length(obs$variable))
+      warning("Provided `scale` vector not same length as `variable` vector.")
+    }
+    tmp <- c()
+    for(i in 1:length(obs$variable)) {
+      str_else <- ifelse(length(obs$variable)-i <= 0, str_else <- "else", "else if")
+      str_if <- ifelse(i == 1, "if", "  ")
+      tmp <- c(
+        tmp,
+        paste0("      ", str_if, " (obs_type[i+1]==", i, ") { obs.insert(obs.end(), ", obs$variable[i], "/(", obs$scale[i], ")); } ", str_else, " ")
+      )
```

Comment:
```suggestion
        paste0(
          "      ", str_if, " (obs_type[i+1]==", i, ")", 
          " { obs.insert(obs.end(), ", obs$variable[i], "/(", obs$scale[i], ")); } ", 
          str_else, " "
        )
      )
```
adding line breaks like this might make it a bit easier to read - tried to mimic how line breaks might look if I were writing it in code.